astropy/io/fits/convenience.py

```diff
@@ -489,7 +514,9 @@ def table_to_hdu(table):
 
             col.null = fill_value.astype(table[col.name].dtype)
     else:
-        table_hdu = BinTableHDU.from_columns(np.array(table.filled()))
+        table_hdu = BinTableHDU.from_columns(np.array(table.filled()), header=hdr)
+
+    coord_meta = table.meta.pop('__coordinate_columns__', None)
```

Comment:
@mhvk I actually did so looking at the ECSV PR. I think I saw ``__mixin_columns__`` there. But, since time is a coordinate column and not a mixin in FITS (as you stated) and also since not all mixins are included (Quantity is a mixin column but won't be in this meta) and probably we would further extend this to SkyCoord, I thought ``__coordinate_columns__`` would be better.

I could store this in the meta of individual columns. I actually did think of doing so, but then I thought that if it is in the table meta, I can directly check ``if coord_meta:`` and then only invest time in this instead of checking each column's meta.

Though I think you would have a better implementation in mind so let me know!